ReactWindows/ReactNative.Net46/Bridge/DispatcherHelpers.cs

```diff
@@ -17,12 +18,47 @@ public static void AssertOnDispatcher()
 
         public static bool IsOnDispatcher()
         {
-            return Thread.CurrentThread == Dispatcher.CurrentDispatcher.Thread;
+            var dispatcher = Application.Current != null
+                ? Application.Current.Dispatcher
+                : Dispatcher.CurrentDispatcher;
+
+            return dispatcher.CheckAccess();
         }
 
-        public static async void RunOnDispatcher(Action action)
+        public static async void RunOnDispatcher(Action action, bool runAsync = true)
         {
-            await Dispatcher.CurrentDispatcher.InvokeAsync(action).Task.ConfigureAwait(false);
+            if (action == null)
+            {
+                throw new InvalidOperationException("No Action");
+            }
+
+            var dispatcher = Application.Current != null
+                ? Application.Current.Dispatcher
+                : Dispatcher.CurrentDispatcher;
+
+
+            if (dispatcher.CheckAccess())
+            {
+                if (runAsync)
+                {
+                    await dispatcher.InvokeAsync(action).Task.ConfigureAwait(false);
+                }
+                else
+                {
+                    action.Invoke();
+                }
+            }
```

Comment:
I have reverted the implementation of the DispatcherHelpers back to the original content with the set dispatcher explicitly change mentioned elsewhere